gcp-gke/main.tf

```diff
@@ -1,191 +1,207 @@
+  master_auth { #tfsec:ignore:google-gke-no-legacy-authentication - False positive?
+    client_certificate_config {
+      issue_client_certificate = true
     }
+  }
 
-    google-beta = {
-      version = ">= 3.0.0"
-      source  = "hashicorp/google-beta"
+  master_authorized_networks_config {
+    cidr_blocks {
+      cidr_block = "0.0.0.0/0" #tfsec:ignore:google-gke-no-public-control-plane - Public access currently needed for development purposes (see https://linear.app/honestbank/issue/DEVOP-746/fix-ckv-gcp-18-for-terraform-gcp-gke)
+    }
+  }
+
+  workload_identity_config {
+    workload_pool = "${data.google_project.service_project.project_id}.svc.id.goog"
+  }
+
+  node_config {
+    image_type   = "COS_CONTAINERD"
+    machine_type = var.machine_type
+
+    workload_metadata_config {
+      mode = "GKE_METADATA"
     }
 
-    helm = {
-      # Use provider with Helm 3.x support
-      version = "~> 1.3.0"
+    shielded_instance_config {
+      enable_secure_boot          = true
+      enable_integrity_monitoring = true
     }
+  }
 
-    kubernetes = {
-      version = "~> 1.13"
+  addons_config {
+    horizontal_pod_autoscaling {
+      disabled = false
     }
 
-    null = {
-      version = ">=2.1.2, <= 3.0"
-      source  = "hashicorp/null"
+    http_load_balancing {
+      disabled = false
     }
 
-    random = {
-      version = "~> 3.0"
+    network_policy_config {
+      disabled = true
     }
 
-    template = {
-      version = "~> 2.2"
+    istio_config {
+      disabled = true
     }
   }
-}
 
-provider "null" {
-}
+  # Autoscaling/Node Auto-Provisioning
+  cluster_autoscaling {
+    enabled = true
 
-provider "random" {
-}
+    resource_limits {
+      resource_type = "cpu"
+      minimum       = 32
+      maximum       = 320
+    }
 
-provider "template" {
-}
+    resource_limits {
+      resource_type = "memory"
+      minimum       = 128
+      maximum       = 1280
+    }
+  }
 
-resource "random_id" "run_id" {
-  byte_length = 4
-}
+  # Intranode visibility configures networking on each node in the cluster so that traffic sent from one Pod to another
+  # Pod is processed by the cluster's Virtual Private Cloud (VPC) network, even if the Pods are on the same node.
+  # https://cloud.google.com/kubernetes-engine/docs/how-to/intranode-visibility
+  enable_intranode_visibility = true
 
-# Shared VPC Permissions
-data "google_project" "service_project" {
-  provider = google.compute
-}
+  network = var.shared_vpc_self_link
+  # Setting VPC_NAME requires the ip_allocation_policy block
+  networking_mode = "VPC_NATIVE"
+  subnetwork      = var.subnetwork_self_link
+  ip_allocation_policy {
+    # Secondary ranges can be built by https://github.com/honestbank/terraform-gcp-vpc
+    cluster_secondary_range_name  = var.pods_ip_range_name
+    services_secondary_range_name = var.services_ip_range_name
+  }
 
-data "google_project" "host_project" {
-  provider = google.vpc
-}
+  network_policy {
+    enabled = true
```

Comment:
Add comment - I don't think this does anything unless actual NetworkPolicy objects are deployed in the cluster... need to check/link docs.

Reply:
It is a toggle so that the `NetworkPolicy` k8s API can actually function.  There is a note about mixing this with workload identity that we need to be aware of and be sure it's working properly with this enabled, will need to add a global policy for reaching the metadata server if it's broken.

[docs here](https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy#limitations_and_requirements)